### PR TITLE
Fix Rust docs build

### LINKED ISSUE #1270: Rust docs for 0.13 failed to build
### Issue description
[The docs](https://docs.rs/crate/mediasoup/latest) for release 0.13 of mediasoup for rust failed to build. [Here's](https://docs.rs/crate/mediasoup/0.13.0/builds/1036978) a direct link to the error.

I can't really try and help more in depth at the moment, but I do know you can try building docs locally to test the build https://github.com/rust-lang/docs.rs#build-subcommand.

### PR DESCRIPTION
**Pull request by ibc**
Fixes #1270

### CODE REVIEW

**Review by nazar-pc — APPROVED** [review]

**Comment by ibc** [comment]
Just wondering if `cargo doc` should be added to CI, I'm adding it.

**Comment by nazar-pc** [comment]
That alone will not fix this particular issue, `DOCS_RS` is an environment variable only set on workers that build documentation on docs.rs, you can set that environment variable in CI though.

I'd recommend this (+`DOCS_RS`): https://github.com/subspace/subspace/blob/1b508553239dec73b88872146c8df68f788bbfd2/.github/workflows/rust.yml#L134-L137

**Comment by ibc** [comment]
Ok, how good is this?

```
DOC_RS=true cargo doc
warning: unresolved link to `WebRtcServer`
  --> rust/src/data_structures.rs:48:43
   |
48 | /// Listening protocol, IP and port for [`WebRtcServer`] to listen on.
   |                                           ^^^^^^^^^^^^ no item named `WebRtcServer` in scope
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
   = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default

warning: unresolved link to `crate::data_consumer::DataConsumer::on_data_producer_resume`
   --> rust/src/router/data_producer.rs:474:57
    |
474 |     /// Calls [`DataConsumer::on_data_producer_resume`](crate::data_consumer::DataConsumer::on_data_producer_resume)
    |                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the enum `DataConsumer` has no variant or associated item named `on_data_producer_resume`

warning: `mediasoup` (lib doc) generated 2 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 0.34s


echo $?
0
```

**Comment by nazar-pc** [comment]
Yeah, you'll have to fix warnings now. Those are actual links to other data structures. You can replace
> [``]

with just
> ``

So it is not a link anymore. Or make it a proper link like in a second warning, but make sure path to data structure is correct.

**Comment by ibc** [comment]
Wow, I've found a bug. I mean, docs did.

**Comment by ibc** [comment]
How is possible that this fix (I didn't change ANYTHING else) doesn't make `cargo build` to fail?

```diff
     /// Callback is called when the associated data producer is resumed.
-    pub fn on_producer_resume<F: Fn() + Send + Sync + 'static>(&self, callback: F) -> HandlerId {
+    pub fn on_data_producer_resume<F: Fn() + Send + Sync + 'static>(&self, callback: F) -> HandlerId {
         self.inner()
             .handlers
             .data_producer_resume
```

**Comment by nazar-pc** [comment]
It is just a public method name, could be anything :man_shrugging:

**Comment by ibc** [comment]
cargo is wrong here:

```
DOC_RS=1 RUSTDOCFLAGS="-D rustdoc::broken-intra-doc-links -D rustdoc::private_intra_doc_links" cargo doc --locked --all --no-deps --lib

 Documenting mediasoup v0.13.0 (/Users/ibc/src/v3-mediasoup/rust)
error: unresolved link to `WebRtcServer`
  --> rust/src/data_structures.rs:48:59
   |
48 | ...`WebRtcServer`][`WebRtcServer`](crate::webrtc_server::WebRtcServer) to listen on.
   |                     ^^^^^^^^^^^^ no item named `WebRtcServer` in scope
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
   = note: requested on the command line with `-D rustdoc::broken-intra-doc-links`

error: could not document `mediasoup`
```

**Comment by nazar-pc** [comment]
It isn't, looks like you have this:
> [`WebRtcServer`][`WebRtcServer`]()

Instead of this:
> [`WebRtcServer`]()

**Comment by ibc** [comment]
OMG

**Comment by ibc** [comment]
PR ready. Yes, an unrelated change in `Cargo.lock` but I'm happy.

File: rust/src/router/data_consumer.rs [hunk 1]
```diff
@@ -903,7 +903,7 @@ impl DataConsumer {
     }
 
     /// Callback is called when the associated data producer is resumed.
-    pub fn on_producer_resume<F: Fn() + Send + Sync + 'static>(&self, callback: F) -> HandlerId {
+    pub fn on_data_producer_resume<F: Fn() + Send + Sync + 'static>(&self, callback: F) -> HandlerId {
```
Comment (nazar-pc) [inline rust/src/router/data_consumer.rs]: This is a breaking change BTW. Either we should make a minor version bump or keep the old name alongside the new name.

**Review by nazar-pc — COMMENTED** [review]
And `cargo fmt` will not hurt

File: rust/src/router/data_consumer.rs (hunk 1 above)
Comment (ibc) [reply rust/src/router/data_consumer.rs]: I'm pretty sure nobody noticed it. But we can release next version later. Will do.

Comment (nazar-pc) [reply rust/src/router/data_consumer.rs]: I know, but is that this is a breaking change that will potentially make code fail to compile, we should honor that

Comment (ibc) [reply rust/src/router/data_consumer.rs]: See https://github.com/versatica/mediasoup/pull/1271/commits/b36e52556e55d69907edbd0da999e1f37346e463

Comment (ibc) [reply rust/src/router/data_consumer.rs]: https://github.com/versatica/mediasoup/pull/1272

**Pull request merged**